pinot-common/src/main/java/org/apache/pinot/common/request/context/RequestContextUtils.java

```diff
@@ -111,11 +111,9 @@ public static FilterContext getFilter(Expression thriftExpression) {
         return getFilter(thriftFunction);
       case IDENTIFIER:
         // Convert "WHERE a" to "WHERE a = true"
-        return new FilterContext(FilterContext.Type.PREDICATE, null,
-            new EqPredicate(getExpression(thriftExpression), getStringValue(RequestUtils.getLiteralExpression(true))));
+        return FilterContext.forPredicate(new EqPredicate(getExpression(thriftExpression), "true"));
       case LITERAL:
-        // TODO: Handle literals.
-        throw new IllegalStateException();
+        return FilterContext.forConstant(new LiteralContext(thriftExpression.getLiteral()).getBooleanValue());
```

Comment:
general question: why do we need to repeat ourselves on these utils one for thrift and one for function context?

it looks like the only difference is one is for WHERE and one is for HAVING?

Reply:
One is used in aggregation FILTER where the expression is already parsed into context format